lib/foreman/gettext/support.rb

```diff
@@ -42,19 +42,29 @@ def self.add_text_domain(locale_domain, locale_dir)
           :report_warning => false
       end
 
-      def self.register_human_localenames
-        original_locale = FastGettext.locale
-        FastGettext.class.class_eval { attr_accessor :human_available_locales }
-        FastGettext.human_available_locales = []
-        FastGettext.default_available_locales.sort.each do |locale|
-          FastGettext.locale = locale
-          # TRANSLATORS: Provide locale name in native language (e.g. English, Deutsch or Portugues)
-          human_locale = _("locale_name")
-          human_locale = locale if human_locale == "locale_name"
-          FastGettext.human_available_locales << [human_locale, locale]
+      # Loading all available locales to get the language translation is slow
+      # (2 seconds for 20 languages), so let's lazy load that.
+      FastGettext.class.class_eval do
+        attr_accessor :mutex, :locales
+
+        def human_available_locales
+          original_locale = FastGettext.locale
+          mutex.synchronize do
+            return locales if locales
+            Rails.logger.debug "Loading and caching all available locales"
+            locales = []
+            FastGettext.default_available_locales.sort.each do |locale|
+              FastGettext.locale = locale
+              # TRANSLATORS: Provide locale name in native language (e.g. English, Deutsch or Portugues)
```

Comment:
Google translate suggests it's
```suggestion
              # TRANSLATORS: Provide locale name in native language (e.g. English, Deutsch or Português)
```

However, I also notice it was already that way. Maybe to avoid UTF-8 in code files?

Reply:
Reworded.